### ISSUE TITLE
Unable to download from itv.com

### ISSUE THREAD

**Issue by ThudnBlundr** [ytdl-org/youtube-dl#27582]
## Checklist

- [x ] I'm reporting a broken site support
- [ x] I've verified that I'm running youtube-dl version **2020.12.29**
- [x ] I've checked that all provided URLs are alive and playable in a browser
- [x ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x ] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
PASTE VERBOSE LOG HERE
```
[debug] System config: []
[debug] User config: ['-o', 'F:/8Tb-Videos/_For-Processing/%(title)s.%(ext)s', '--ffmpeg-location', 'C:/Progra~2/get_iplayer/utils']
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.itv.com/hub/bugsy-malone/CFD0085']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2020.12.29
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.19041
[debug] exe versions: ffmpeg 4.2.1
[debug] Proxy map: {}
[debug] Using fake IP 25.181.70.25 (GB) as X-Forwarded-For.
[ITV] CFD0085: Downloading webpage
[ITV] CFD0085: Downloading JSON metadata
[ITV] CFD0085: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp581_vk6o\build\youtube_dl\YoutubeDL.py", line 803, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp581_vk6o\build\youtube_dl\YoutubeDL.py", line 824, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp581_vk6o\build\youtube_dl\extractor\common.py", line 532, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp581_vk6o\build\youtube_dl\extractor\itv.py", line 115, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp581_vk6o\build\youtube_dl\extractor\common.py", line 1367, in _sort_formats
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

## Description

WRITE DESCRIPTION HERE

I had a similar problem a few days ago, but my version was a week or two out of date. Updating to the latest version fixed it until today.

**Comment by dstftw** [comment]
Video 32 has restricted access.

**Comment by Vangelis66** [comment]
@ThudnBlundr 

Happy New Year! :smile: 
This is quite a weird bug you've reported! 
At first I, probably like @dstftw, thought you are probably being geofenced 
at the stream's CDN door, a conclusion consistent with: 
```
ITV] CFD0085: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
```
For starters though, the correct ID of the programme (film) is `CFD0085a0001`; when I tried myself 
with a **whitelisted** UK VPN, I got mixed results; the `-F` option worked: 
```
youtube-dl -F "https://www.itv.com/hub/bugsy-malone/CFD0085a0001"

[ITV] CFD0085a0001: Downloading webpage
[ITV] CFD0085a0001: Downloading JSON metadata
[ITV] CFD0085a0001: Downloading m3u8 information
[info] Available formats for CFD0085a0001:
format code  extension  resolution note
hls-136      mp4        audio only  136k , mp4a.40.2@128k
hls-638      mp4        512x288     638k , avc1.64001F@ 473k, 50.0fps, mp4a.40.2
@128k
hls-846      mp4        512x288     846k , avc1.64001F@ 669k, 50.0fps, mp4a.40.2
@128k
hls-1201     mp4        896x504    1201k , avc1.64001F@1004k, 50.0fps, mp4a.40.2
@128k
hls-1513     mp4        896x504    1513k , avc1.64001F@1299k, 50.0fps, mp4a.40.2
@128k
hls-1829     mp4        896x504    1829k , avc1.64001F@1596k, 50.0fps, mp4a.40.2
@128k (best)
```
but the actual download attempt failed, consistent with your report: 
```
youtube-dl --console-title --hls-prefer-native --hls-use-mpegts -c --no-part "https://www.itv.com/hub/bugsy-malone/CFD0085a0001"

[ITV] CFD0085a0001: Downloading webpage
[ITV] CFD0085a0001: Downloading JSON metadata
[ITV] CFD0085a0001: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug
 . Make sure you are using the latest version; type  youtube-dl -U  to update. B
e sure to call youtube-dl with the --verbose flag and include its complete output.
```
Until this has been properly addressed by the devs (sorry, I don't "speak" Python), 
I can offer a workaround: 
1. Use the `-j` flag and redirect `yt-dl`'s output to a file, e.g. 

`youtube-dl "https://www.itv.com/hub/bugsy-malone/CFD0085a0001" -j > j.txt`

2. As swiftly as you best can, open the `j.txt` file with a proper text editor, 
then copy the (last) manifest URL, i.e. 
```
https://itvpnpdotcom.content.itv.com/CFD0085-0001-005/18/5/VAR012/CFD0085-0001-005_18_5_VAR012.ism/.m3u8?hdnea=st=1609465782~exp=1609487382~acl=/CFD0085-0001-005/*~data=nohubplus~hmac=ad097a5403ab4baf597e8c0fd95a9fc137fa77aa41ef1170c5b6b7291aed24fd
```
NB: The authentication token is very short-lived, that's why you have to act quickly...
If you feed that manifest directly to yt-dl, download will succeed: 

```
youtube-dl --console-title --hls-prefer-native --hls-use-mpegts -c --no-part "https://itvpnpdotcom.content.itv.com/CFD0085-0001-005/18/5/VAR012/CFD0085-0001-005_18_5_VAR012.ism/.m3u8?hdnea=st=1609465782~exp=1609487382~acl=/CFD0085-0001-005/*~data=nohubplus~hmac=ad097a5403ab4baf597e8c0fd95a9fc137fa77aa41ef1170c5b6b7291aed24fd" -o "ITV3_BugsyMalone_202012281050[CFD0085a0001].mp4

[generic] *~data=nohubplus~hmac=ad097a5403ab4baf597e8c0fd95a9fc137fa77aa41ef1170
c5b6b7291aed24fd: Requesting header
[generic] *~data=nohubplus~hmac=ad097a5403ab4baf597e8c0fd95a9fc137fa77aa41ef1170
c5b6b7291aed24fd: Downloading m3u8 information
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 892
[download] Destination: ITV3_BugsyMalone_202012281050[CFD0085a0001].mp4
[download]   0.9% of ~1.04GiB at 597.19KiB/s ETA 44:16
```
Best wishes!

**Comment by ThudnBlundr** [comment]
Thanks for that. I'll get on it this afternoon. I didn't mean to open a can of worms!

**Comment by remitamine** [comment]
https://github.com/ytdl-org/youtube-dl/issues/27465#issuecomment-747441666

**Comment by Vangelis66** [comment]
@remitamine

Happy New Year to you, too! :santa: 

I have read the _issue comment_ you linked to, in fact I'm always keeping an eye on **ITV related** issues; 
as a further test, using the same **whitelisted UK VPN** (its **residential** IP address is roughly placed in **Aberdeen**, Scotland), I performed **ten consecutive**, with 3min intervals between them, runs of the command: 
```
youtube-dl --console-title --hls-prefer-native --hls-use-mpegts -c --no-part "https://www.itv.com/hub/bugsy-malone/CFD0085a0001"
```
and I have **always** gotten the dreaded **403** response from the stream CDN server: 
```
[ITV] CFD0085a0001: Downloading webpage
[ITV] CFD0085a0001: Downloading JSON metadata
[ITV] CFD0085a0001: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
```
After that, I performed  **ten consecutive**, with 3min intervals between them, runs of the command: 
```
youtube-dl "https://www.itv.com/hub/bugsy-malone/CFD0085a0001" -j > j.txt
```
and **not once** did I find manifest urls originating from the **Amazon S3** CDN (`itvpnpdotcom.cdn1.content.itv.com`); I'm attaching a file created mere minutes ago: [j.txt](https://github.com/ytdl-org/youtube-dl/files/5759105/j.txt)
FTR, I have no issues downloading non-film content from ITV via the same VPN... 
This whole thing definitely smells of fish... :grey_question:

**Comment by baffled-mart** [comment]
Appreciate this maybe irrelevant but...

During the past few weeks I have also been getting sporadic " WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden" on ITV downloads.  Eventually the required programme would download.  No diagnostics like Vangelis66, just repetitive attempts until successful.

If it happens again I'll add the relevant output.

Location UK.
OS Vista
yt-dl versions various

**Comment by baffled-mart** [comment]
Just had same problem with
https://www.itv.com/hub/and-finally-with-trevor-mcdonald-w-t/10a0669a0001
Proceeded per instructions from Vangelis66 (including his parameters which are unknown to me) and have success.
But failure when using my original params.

Failed run
V:\youtube-dl_2021.01.03>youtube-dl -f mp4 --no-check-certificate --ignore-errors --restrict-filenames --download-archive youtube-dl-history.txt --add-metadata --verbose "https://www.itv.com/hub/and-finally-with-trevor-mcdonald-w-t/10a0669a0001"
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-f', 'mp4', '--no-check-certificate', '--ignore-errors', '--restrict-filenames', '--download-archive', 'youtube-dl-history.txt', '--add-metadata', '--verbose', 'https://www.itv.com/hub/and-finally-with-trevor-mcdonald-w-t/10a0669a0001']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2021.01.03
[debug] Python version 3.4.4 (CPython) - Windows-Vista-6.0.6002-SP2
[debug] exe versions: ffmpeg N-80923-g50747d6, ffprobe N-80923-g50747d6, rtmpdump 2.4
[debug] Proxy map: {}
[debug] Using fake IP 25.209.182.245 (GB) as X-Forwarded-For.
[ITV] 10a0669a0001: Downloading webpage
[ITV] 10a0669a0001: Downloading JSON metadata
[ITV] 10a0669a0001: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 803, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 824, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\extractor\common.py", line 532, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\extractor\itv.py", line 115, in _real_extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\extractor\common.py", line 1367, in _sort_formats
youtube_dl.utils.ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

Failed with mixture mine/Vangelis66
V:\youtube-dl_2021.01.03>youtube-dl -f mp4 --no-check-certificate --ignore-errors --restrict-filenames --download-archive youtube-dl-history.txt --add-metadata --verbose "https://itvpnpdotcom.content.itv.com/10-0669-0001-001/18/1/VAR012/10-0669-0001-001_18_1_VAR012.ism/.m3u8?hdnea=st=1609722448~exp=1609744048~acl=/10-0669-0001-001/*~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4"
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-f', 'mp4', '--no-check-certificate', '--ignore-errors', '--restrict-filenames', '--download-archive', 'youtube-dl-history.txt', '--add-metadata', '--verbose', 'https://itvpnpdotcom.content.itv.com/10-0669-0001-001/18/1/VAR012/10-0669-0001-001_18_1_VAR012.ism/.m3u8?hdnea=st=1609722448~exp=1609744048~acl=/10-0669-0001-001/*~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2021.01.03
[debug] Python version 3.4.4 (CPython) - Windows-Vista-6.0.6002-SP2
[debug] exe versions: ffmpeg N-80923-g50747d6, ffprobe N-80923-g50747d6, rtmpdump 2.4
[debug] Proxy map: {}
[generic] *~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4: Requesting header
[generic] *~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4: Downloading m3u8 information
[debug] Invoking downloader on 'https://itvpnpdotcom.content.itv.com/10-0669-0001-001/18/1/VAR012/10-0669-0001-001_18_1_VAR012.ism/10-0669-0001-001_18_1_VAR012-audio=128000-video=1517658.m3u8'
[download] Destination: m3u8-_~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4.mp4
[debug] ffmpeg command line: ffmpeg -y -loglevel verbose -headers "Cookie: hdntl=exp=1609809031~acl=%2f10-0669-0001-001%2f*~data=hdntl,nohubplus~hmac=0517b48d2ee2264d6a73eefb96c30fdbeacf57fa3a9f2f2c3d882f2613ea01a5
Accept-Encoding: gzip, deflate
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3635.2 Safari/537.36
Accept-Charset: ISO-8859-1,utf-8;q=0.7,*;q=0.7
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-us,en;q=0.5
" -i "https://itvpnpdotcom.content.itv.com/10-0669-0001-001/18/1/VAR012/10-0669-0001-001_18_1_VAR012.ism/10-0669-0001-001_18_1_VAR012-audio=128000-video=1517658.m3u8" -c copy -f mp4 "-bsf:a" aac_adtstoasc "file:m3u8-_~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4.mp4.part"
ffmpeg version N-80923-g50747d6 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.4.0 (GCC)
  configuration: --enable-gpl --enable-version3 --disable-w32threads --enable-dxva2 --enable-libmfx --enable-nvenc --enable-avisynth --enable-bzlib --enable-fontconfig --enable-frei0r --enable-gnutls --enable-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-libmodplug --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-librtmp --enable-libschroedinger --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwavpack  --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable-lzma --enable-decklink --enable-zlib
  libavutil      55. 28.100 / 55. 28.100
  libavcodec     57. 48.103 / 57. 48.103
  libavformat    57. 41.100 / 57. 41.100
  libavdevice    57.  0.102 / 57.  0.102
  libavfilter     6. 47.100 /  6. 47.100
  libswscale      4.  1.100 /  4.  1.100
  libswresample   2.  1.100 /  2.  1.100
  libpostproc    54.  0.100 / 54.  0.100
[tls @ 002bf060] Creating security context failed (0x80090302)
https://itvpnpdotcom.content.itv.com/10-0669-0001-001/18/1/VAR012/10-0669-0001-001_18_1_VAR012.ism/10-0669-0001-001_18_1_VAR012-audio=128000-video=1517658.m3u8: Unknown error occurred


ERROR: ffmpeg exited with code 1
  File "__main__.py", line 19, in <module>
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\__init__.py", line 474, in main
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\__init__.py", line 464, in _real_main
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 2029, in download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 796, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 803, in wrapper
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 835, in __extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 869, in process_ie_result
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 1654, in process_video_result
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 1936, in process_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 1875, in dl
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\downloader\common.py", line 366, in download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\downloader\external.py", line 64, in real_download
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\downloader\common.py", line 165, in report_error
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 625, in report_error
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7n_m4vnl\build\youtube_dl\YoutubeDL.py", line 587, in trouble

Success using Vangelis66 params.
V:\youtube-dl_2021.01.03>youtube-dl --console-title --hls-prefer-native --hls-use-mpegts -c --no-part "https://itvpnpdotcom.content.itv.com/10-0669-0001-001/18/1/VAR012/10-0669-0001-001_18_1_VAR012.ism/.m3u8?hdnea=st=1609722448~exp=16097440
48~acl=/10-0669-0001-001/*~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4" -o "and-finally-with-trevor-mcdonald-w-t_10a0669a0001.mp4"
[generic] *~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4: Requesting header
[generic] *~data=nohubplus~hmac=4c378676e80b892bfbae60f45daac2303e10bb96334272d32d0cef4630d9f5a4: Downloading m3u8 information
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 467
[download] Destination: and-finally-with-trevor-mcdonald-w-t_10a0669a0001.mp4
[download]   1.5% of ~561.68MiB at 784.82KiB/s ETA 17:53

**Comment by baffled-mart** [comment]
Forgot to remove url from batch file and now downloading successfully.
Just love transient problems!

V:\youtube-dl_2021.01.03>youtube-dl -f mp4 --no-check-certificate --ignore-errors --restrict-filenames --downloae youtube-dl-history.txt --add-metadata --batch-file videos.txt
[ITV] 10a0669a0001: Downloading webpage
[ITV] 10a0669a0001: Downloading JSON metadata
[ITV] 10a0669a0001: Downloading m3u8 information
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 467
[download] Destination: And_Finally..._with_Trevor_Mcdonald_W-t_-_Series_1_-_Episode_1-10a0669a0001.mp4
[download]   3.7% of ~556.06MiB at 862.01KiB/s ETA 21:53

**Comment by Vangelis66** [comment]
@baffled-mart wrote: 

> Proceeded per instructions from @Vangelis66 (including **his parameters which are unknown to me**) 
and have success.

... All used **options** are described/detailed in `yt-dl`'s HELP file; 
you can generate a copy of it on disk via: 

`youtube-dl -h > Help.txt 2>&1`

Cheers :smile:

**Comment by britmotley** [comment]
can someone explain exactly how to set Vangelis66 params. :-)

**Comment by Vangelis66** [comment]
@britmotley 

Haven't I been _verbose_ enough for you?

https://github.com/ytdl-org/youtube-dl/issues/27582#issuecomment-753243535

**Comment by britmotley** [comment]
@Vangelis66 
Al I can tell you is that my ITV downloads crash after a few minutes, then I retry and they crash again,and it just goes on and on. 
Or it says there are no streams available
I'm simply not as DOS savvy as you guys on here :-(

**Comment by torbs723** [comment]
@britmotley
When you say the downloads crash, what exactly do you mean? Do you mean they start to actually download successfully and then stop and fail? Or do you mean you get the usual and then 403 forbidden/No video formats found?

**Comment by Vangelis66** [comment]
@torbs723 asked: 

> @Vangelis66
So are **you** saying that the geoblocking on the Amazon S3 CDN 
is blocking lots of legitimate users?

If you read from the start of this thread, you'll discover that the **Amazon S3 CDN** (hostname = itvpnpdotcom.cdn1.content.itv.com) was first mentioned by yt-dl's main developer **remitamine** in another ITV-related issue: 

> when the manifest is served from:
(snipped)
itvpnpdotcom.cdn1.content.itv.com **fails for whatever reason** (Amazon S3).

Therefore, **I** am not saying _anything_ myself... I don't fetch from ITV routinely, thus in my sporadic downloads I haven't seen concrete evidence of referenced CDN... But the devs surely know better 😉 

It appears that ITV staff are now being more aggressive in their **ACL**s (Access-Control-Lists), IOW **geo-filters**, perhaps **even within the UK itself**... I know, for example, that Scottish IPs might be blacklisted, because of the existence of local broadcaster STV...
... And IPs from known datacenters within the UK (ones used by commercial VPNs/DNS/proxies etc) are **routinely blacklisted** in a blanket fashion, not only in the "Playlist" stage (which `yt-dl`, most of the times, manages to circumvent) but also in the actual stream CDNs (recent occurrence 😠 ).

> is there a way at the moment to force yt-dl to exclude Amazon S3 playlists?

Without being a coder myself, as I see it it's `yt-dl` querying the ITV **playlist API**, which then responds back with a stream CDN in a fashion not controlled by `yt-dl` itself`... If your **exposed UK IP** is blocked by the CDN, you have to start all over again...

... And to think there were better times in the not-so-distant past when the stream CDN(s) would not geo-fence at all... Alas, all good things don't last for long... What I'm now dreading is the very probable event those last (DRM-free) ITV HLSe streams (targeting older iOS devices) are decommissioned, then it'll be game over, sadly... 😢

**Comment by torbs723** [comment]
> @torbs723 asked:

> It appears that ITV staff are now being more aggressive in their **ACL**s (Access-Control-Lists), IOW **geo-filters**, perhaps **even within the UK itself**... I know, for example, that Scottish IPs might be blacklisted, because of the existence of local broadcaster STV...
> ... And IPs from known datacenters within the UK (ones used by commercial VPNs/DNS/proxies etc) are **routinely blacklisted** in a blanket fashion, not only in the "Playlist" stage (which `yt-dl`, most of the times, manages to circumvent) but also in the actual stream CDNs (recent occurrence  ).

Seems like a bad idea though for ITV to start blocking actual UK mainland IPs on one of their servers though? Those are the people they should NOT be blocking from the service. Not seen any problems on the streaming side so far relating to this though. I'm guessing when you play in a browser or on a TV, it goes for the Akamai servers.

**Comment by Vangelis66** [comment]
> Not seen any problems **on the streaming side** so far relating to this though. 
> I'm guessing when **you play in a browser or on a TV, it goes for the Akamai servers.**

_Totally unrelated_ to `yt-dl` 😜 ; **desktop browsers** and **Smart TV**s are being served **DRM**'ed cenc MPEG-DASH[/FairPlay HLS] streams, while, as _**I already wrote**_, `yt-dl` accesses the **only source** of non-DRM'ed streams, **HLSe streams targeting OLD iPhones/iPads**...

**Comment by torbs723** [comment]
Fair enough. But what I meant was that as I understand it, Amazon S3 is geoblocking the JSON playlists and I would expect, (even in browser streams and Smary TVs are getting DRM'ed streams) that those devices still look to Amazon S3 servers to get the playlists that they need too. So as it's all happening over the same IP, if yt-dl is being refused access to a playlist from Amazon S3, then why wouldn't a browser or Smart TV be blocked from getting their playlists from that server too?

**Comment by Vangelis66** [comment]
> as I understand it, **Amazon S3 is geoblocking the **JSON** playlists**

Not quite...
In `yt-dl`, the **JSON playlists** are being served by the `magni.itv.com` playlist API, which is indeed geo-fenced, but this is usually taken care of by `yt-dl`'s native mechanism; the API (nothing to do itself with Akamai or Amason S3) will return **HLSe master.m3u8 URIs** served from Akamai (mostly): 

> [ITV] CFD0085: Downloading JSON metadata
[ITV] CFD0085: Downloading m3u8 information
WARNING: Failed to download m3u8 information: HTTP Error 403: Forbidden

In OP's log, **JSON metadata** will successfully be fetched, the 403 occurs when trying to fetch the **m3u8** URI...

> I would expect, (even if browsers and Smart TVs are getting DRM'ed streams) that **those devices still look to Amazon S3 servers to get the [HLS] playlists that they need too**. So as it's all happening over the same IP, if yt-dl is being refused access to a playlist from Amazon S3, then why wouldn't a browser or Smart TV be blocked from getting their playlists from that server too?

As I don't reside in the UK and don't own a **UK Smart TV**, I couldn't possibly tell what exact connections are being made _there_...
Do you have evidence of your UK browser and/or UK Smart TV being delivered streams from `itvpnpdotcom.cdn1.content.itv.com` ? 
Troubleshooting in a desktop browser would be much easier, log-in to your ITVHub account, launch Web Console in Firefox and load an ITV VOD (will use cenc mpeg-dash streams that will require the WidevineCDM); then monitor exchanged web traffic... 

But, IMHO, this issue is being derailed... We should stick to connections between `yt-dl` and ITVHub APIs/CDNs 😉 ...
Bottomline? I don't think much (if anything) can be done within `yt-dl` to deal with refused connections caused by the ever more strict ITVHub's geo-blocking schemes...

**Comment by torbs723** [comment]
> In OP's log, **JSON metadata** will successfully be fetched, the 403 occurs when trying to fetch the **m3u8** URI...
Indeed it **IS** the m3u8 fetch that fails, not the JSON fetch. This is exactly the same issue I'm having where the JSON comes down successfully but the m3u8 fails.

> Do you have evidence of your UK browser and/or UK Smart TV being delivered streams from `itvpnpdotcom.cdn1.content.itv.com` ?
> Troubleshooting in a desktop browser would be much easier, log-in to your ITVHub account, launch Web Console in Firefox and load an ITV VOD (will use cenc mpeg-dash streams that will require the WidevineCDM); then monitor exchanged web traffic...
I can't work out how to monitor web traffic in the Chrome console. All I've found so far is ITV's .js file. Any ideas?

> But, IMHO, this issue is being derailed... We should stick to connections between `yt-dl` and ITVHub APIs/CDNs
I still think this is sort of relevant though. This is an issue with ITV's geoblocking which on their Amazon S3 server does appear to be blocking UK IP addresses like mine and presumably the OP's and others. This is causing the effect in yt-dl that we are now seeing. Unfortunately though this leaves us in a position where there isn't much yt-dl can do to fix this. We have to rely on ITV not to block private IPs in the UK.

Thanks to the digging that you and remitamine have been doing, we now have a better understanding of the wider issue though which may or may not make it easier to resolve.

**Comment by britmotley** [comment]
> @britmotley
> When you say the downloads crash, what exactly do you mean? Do you mean they start to actually download successfully and then stop and fail? Or do you mean you get the usual and then 403 forbidden/No video formats found?

@torbs723 
The downloads start ok then crash , i re enter the command and they resume and then crash.It can take up to 20 attempts to download a whole show
the errors thrown up are always 403 forbidden/No video formats found  or SSLV3 socket errors

**Comment by 89z** [comment]
Issue should be closed, URL is dead

**Comment by ThudnBlundr** [comment]
That particular URL may be dead after a year and a bit. But it still fails with any other video you try to download from ITV Hub. Try this one for instance: [Carry On... Follow That Camel](https://www.itv.com/hub/carry-on-follow-that-camel/CFD0121a0001)

**Comment by 89z** [comment]
Thanks for the update

**Comment by Vangelis66** [comment]
> Try this one for instance: [Carry On... Follow That Camel](https://www.itv.com/hub/carry-on-follow-that-camel/CFD0121a0001)

Hi @ThudnBlundr 😄 
Please take some time and search recent ITV-related issues 😉 ; you'll hopefully find that: 
1. Current, as in even the **latest** `yt-dl` release `v2021.12.17`, ITV InfoExtractor (itvIE) is **broken** 😞 .
2. A fix for that is actually PR #30266, yet un-merged in the `master` branch...
3. ITVHub are currently blocking web requests by `yt-dl`, a "dirty" workaround being to issue a palatable user agent string in the `yt-dl`  command (I myself use ` --user-agent "Chrome/100"`)
4. ITVHub are being very finicky in their list of sanctioned UK IP pools, make sure your UK IP address is whitelisted in their ACL filters...

If provisos 2+3+4 are met, then your sample URI should download fine (at least it does here): 
```
youtube-dl --console-title --hls-prefer-native --hls-use-mpegts -c --no-part --fixup never --user-agent "Chrome/100" "https://www.itv.com/hub/carry-on-follow-that-camel/CFD0121a0001" --write-sub --convert-subs=srt => 

[ITV] CFD0121a0001: Downloading webpage
[ITV] CFD0121a0001: Downloading JSON metadata
[ITV] CFD0121a0001: Downloading m3u8 information
[ITV] CFD0121a0001: Downloading JSON metadata
[info] Writing video subtitles to: Carry On... Follow That Camel-CFD0121a0001.en.vtt
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 898
[download] Destination: Carry On... Follow That Camel-CFD0121a0001.mp4
[download]   0.4% of ~1.14GiB at 1000.00KiB/s ETA 35:53
```
Kindest regards

**Comment by dirkf** [comment]
Gone:
* Bugsy Malone
* Carry On Follow That Camel

Eg: `HTTP Error 404: Not Found (CFD0121a0001: We're really sorry)`

Working in #30266:
* https://www.itv.com/hub/and-finally-with-trevor-mcdonald-w-t/10a0669a0001
